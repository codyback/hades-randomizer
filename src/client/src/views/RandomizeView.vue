<template>
  <div class="row q-pa-sm">
    <div class="col-12 col-md">
      <randomize-settings />
      <randomize-filter />
    </div>
    <div class="col-12 col-md-5">
      <randomized-items />
    </div>
    <div class="col-12 col-md">
    </div>
  </div>
</template>

<script lang="ts" setup>
import RandomizeSettings from '../components/RandomizeSettings.vue';
import RandomizeFilter from '../components/RandomizeFilter.vue';
import RandomizedItems from '../components/RandomizedItems.vue';
</script>
