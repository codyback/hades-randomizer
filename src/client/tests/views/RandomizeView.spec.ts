import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

import RandomizeView from '../../src/views/RandomizeView.vue';

const mountWith = () => {
  return mount(RandomizeView, {
    global: {
      stubs: [
        'randomize-settings',
        'randomize-filter'
      ],
      plugins: [ createTestingPinia() ],
    }
  });
}

describe('RandomizeView', () => {
  it('renders', () => {
    const wrapper = mountWith();
    expect(wrapper.element).toMatchSnapshot();
  });
});
