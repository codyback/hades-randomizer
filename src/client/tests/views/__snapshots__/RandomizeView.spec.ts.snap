// Vitest Snapshot v1

exports[`RandomizeView > renders 1`] = `
<div
  class="row q-pa-sm"
>
  <div
    class="col-12 col-md"
  >
    <randomize-settings-stub />
    <randomize-filter-stub />
  </div>
  <div
    class="col-12 col-md-5"
  >
    <randomize-settings-stub />
    <randomize-filter-stub />
  </div>
  <div
    class="col-12 col-md"
  >
    <randomize-settings-stub />
    <randomize-filter-stub />
  </div>
</div>
`;
